packages/rum-core/src/domain/resource/resourceUtils.ts

```diff
@@ -140,50 +139,28 @@ export function toValidEntry(entry: RumPerformanceResourceTiming) {
   // RumPerformanceResourceTiming, it will ignore entries from requests where timings cannot be
   // collected, for example cross origin requests without a "Timing-Allow-Origin" header allowing
   // it.
-  if (
-    !areInOrder(
-      entry.startTime,
-      entry.fetchStart,
-      entry.domainLookupStart,
-      entry.domainLookupEnd,
-      entry.connectStart,
-      entry.connectEnd,
-      entry.requestStart,
-      entry.responseStart,
-      entry.responseEnd
-    )
-  ) {
-    return undefined
-  }
-
-  if (!hasRedirection(entry)) {
+  const areCommonTimingsInOrder = areInOrder(
+    entry.startTime,
+    entry.fetchStart,
+    entry.domainLookupStart,
+    entry.domainLookupEnd,
+    entry.connectStart,
+    entry.connectEnd,
+    entry.requestStart,
+    entry.responseStart,
+    entry.responseEnd
+  )
+
+  const areRedirectionTimingsInOrder =
+    !hasRedirection(entry) || areInOrder(entry.startTime, entry.redirectStart, entry.redirectEnd, entry.fetchStart)
```

Comment:
<!-- Nitpicks are small, trivial, but necessary changes. Distinguishing nitpick comments significantly helps direct the reader's attention to comments requiring more involvement.  -->
**🥜 nitpick:** ‏ I think it would be easier to read as:
```suggestion
  const areRedirectionTimingsInOrder = hasRedirection(entry)
    ? areInOrder(entry.startTime, entry.redirectStart, entry.redirectEnd, entry.fetchStart)
    : true
```